test/decl/func/rethrows.swift

```diff
@@ -612,3 +612,20 @@ func rdar_47550715() {
   func foo(_: A<F>? = nil) {} // Ok
   func bar(_: A<F>? = .none) {} // Ok
 }
+
+// SR-14270 - test case for diagnostic note 'because_rethrows_default_argument_throws'
+func nonThrowableDefaultRethrows(_ f: () throws -> () = {}) rethrows {
+  try f()
+}
+// NOTE: under ideal circumstances, this test case would not have any error or diagnostic.
+// However, due to SR-1534, an error and diagnostic is emitted. Once SR-1534 is fixed, the
+// below should compile correctly, and there should no longer be an expected error or note.
```

Comment:
Just a suggestion, maybe we can simplify this by only mentioning why it should work and just referencing the SR

```suggestion
// NOTE: This should compile and not emit a diagnostic because ideally the compiler could statically know the default argument value could never throw. See SR-1524.
```

Reply:
I agree; I didn't really know how to articulate the comments very well. Thank you